src/components/Timer/Timer.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import { Typography } from '@mui/material';
+import { TimeBox } from './styled';
+
+type TimerProps = {
+  time: number;
+  finishedTimer: () => void;
+};
+
+const Timer = ({ time, finishedTimer }: TimerProps) => {
+  const [minutes, setMinutes] = useState(0);
+  const [seconds, setSeconds] = useState(0);
+
+  const timer = (timeInMinutes: number, endTimer: () => void) => {
+    let sec: number = 0;
+    let min: number = timeInMinutes;
+
+    const timerId = setInterval(() => {
+      if (sec === 0) {
+        min -= 1;
+        sec = 59;
+        setMinutes(min);
+      } else {
+        sec -= 1;
+      }
+
+      if (min === 0 && sec === 0) {
+        endTimer();
+        clearInterval(timerId);
+      }
+      setSeconds(sec);
+    }, 1000);
+  };
+
+  useEffect(() => {
+    timer(time, finishedTimer);
```

Comment:
you need to clear setInterval/setTimeout when component will unmount
if user will leave this page interval will continue working